eth/stagedsync/stage_execute.go

```diff
@@ -132,6 +147,9 @@ func SpawnExecuteBlocksStage(s *StageState, stateDB ethdb.Database, chainConfig
 	if _, err := batch.Commit(); err != nil {
 		return fmt.Errorf("sync Execute: failed to write batch commit: %v", err)
 	}
+	if _, err = tx.Commit(); err != nil {
```

Comment:
what will happen when we Commit and the defer calls Rollback?

Reply:
Commit sets internal transaction object to nil, then rollback do something only if that object is not nil. So, rollback after commit must not do anything. Same for double-rollback.

Reply:
With Tx need to be more strict than with mutation: if tx not closed well - it can cause deadlock or leak.